src/System.Slices/System/SpanExtensions_binary.cs

```diff
@@ -62,6 +62,36 @@ public static partial class SpanExtensionsLabs
         }
 
         /// <summary>
+        /// Reads a structure of type T out of a slice of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryRead<[Primitive]T>(this ReadOnlySpan<byte> slice, out T value)
+            where T : struct
+        {
+            if (Unsafe.SizeOf<T>() > (uint)slice.Length) {
+                value = default(T);
+                return false;
+            }
+            value = slice.Cast<byte, T>()[0];
```

Comment:
I have added issue https://github.com/dotnet/corefx/issues/16479 to `ReadUnaligned<T>`/`WriteUnaligned<T>` APIs on the Unsafe class that will make it possible to write this in portable way.